scipy/linalg/tests/test_basic.py

```diff
@@ -40,11 +40,21 @@
 
 from scipy._lib._version import NumpyVersion
 
-REAL_DTYPES = [np.float32, np.float64]
-COMPLEX_DTYPES = [np.complex64, np.complex128]
+REAL_DTYPES = [np.float32, np.float64, np.float128]
+COMPLEX_DTYPES = [np.complex64, np.complex128, np.complex256]
 DTYPES = REAL_DTYPES + COMPLEX_DTYPES
 
 
+def _eps_cast(dtyp):
+    """Get the epsilon for dtype, possibly downcast to BLAS types."""
+    dt = dtyp
+    if dt == np.float128:
+        dt = np.float64
+    elif dt == np.complex256:
```

Comment:
complex256 and float128 do not exist on all platforms (on some the sizes are complex192 and float96 etc.). The "invariant" names are iirc np.longdouble, np.clongdouble